The Travis test fails with the following error:
```
 GO     operator/cilium-operator
# github.com/cilium/cilium/pkg/bgpv1/manager
../pkg/bgpv1/manager/advertisements_reconciler.go:107:6: unknown field Net in struct literal of type "github.com/cilium/cilium/pkg/bgpv1/types".Advertisement
../pkg/bgpv1/manager/advertisements_reconciler.go:107:27: advrt.Net undefined (type "github.com/cilium/cilium/pkg/bgpv1/types".Advertisement has no field or method Net)
make[1]: *** [Makefile:14: cilium-agent] Error 1
make: *** [Makefile:59: daemon] Error 2
```

That does not seem like a flake :(

--- a/pkg/bgpv1/agent/annotations.go
+++ b/pkg/bgpv1/agent/annotations.go
@@ -6,7 +6,7 @@ package agent
 import (
 	"errors"
 	"math"
-	"net"
+	"net/netip"
 	"strconv"
 	"strings"
 
@@ -165,8 +165,8 @@ func parseAnnotation(key string, value string) (int, Attributes, error) {
 		}
 		switch kv[0] {
 		case "router-id":
-			ip := net.ParseIP(kv[1])
-			if ip.IsUnspecified() {
+			addr, _ := netip.ParseAddr(kv[1])
+			if addr.IsUnspecified() {
 				return 0, out, ErrAttrib{key, kv[0], "could not parse in an IPv4 address"}
 			}
 			out.RouterID = kv[1]

--- a/pkg/bgpv1/agent/controller.go
+++ b/pkg/bgpv1/agent/controller.go
@@ -6,7 +6,7 @@ package agent
 import (
 	"context"
 	"fmt"
-	"net"
+	"net/netip"
 	"time"
 
 	"github.com/sirupsen/logrus"
@@ -15,6 +15,7 @@ import (
 
 	"github.com/cilium/cilium/pkg/hive"
 	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/ip"
 	v2alpha1api "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
 	"github.com/cilium/cilium/pkg/k8s/resource"
 	slimlabels "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/labels"
@@ -86,9 +87,27 @@ type ControlPlaneState struct {
 	// control plane is running on.
 	Annotations AnnotationMap
 	// The current IPv4 address of the agent, reachable externally.
-	IPv4 net.IP
+	IPv4 netip.Addr
 	// The current IPv6 address of the agent, reachable externally.
-	IPv6 net.IP
+	IPv6 netip.Addr
+}
+
+// ResolveRouterID resolves router ID, if we have an annotation and it can be
+// parsed into a valid ipv4 address use it. If not, determine if Cilium is
+// configured with an IPv4 address, if so use it. If neither, return an error,
+// we cannot assign an router ID.
+func (cstate *ControlPlaneState) ResolveRouterID(localASN int) (string, error) {
+	if _, ok := cstate.Annotations[localASN]; ok {
+		if parsed, err := netip.ParseAddr(cstate.Annotations[localASN].RouterID); err == nil && !parsed.IsUnspecified() {
+			return parsed.String(), nil
+		}
+	}
+
+	if !cstate.IPv4.IsUnspecified() {
+		return cstate.IPv4.String(), nil
+	}
+
+	return "", fmt.Errorf("router id not specified by annotation and no IPv4 address assigned by cilium, cannot resolve router id for virtual router with local ASN %v", localASN)
 }
 
 type policyLister interface {
@@ -369,12 +388,15 @@ func (c *Controller) Reconcile(ctx context.Context) error {
 		return fmt.Errorf("failed to retrieve Node's pod CIDR ranges: %w", err)
 	}
 
+	ipv4, _ := ip.AddrFromIP(nodeaddr.GetIPv4())
+	ipv6, _ := ip.AddrFromIP(nodeaddr.GetIPv6())
+
 	// define our current point-in-time control plane state.
 	state := &ControlPlaneState{
 		PodCIDRs:    podCIDRs,
 		Annotations: annoMap,
-		IPv4:        nodeaddr.GetIPv4(),
-		IPv6:        nodeaddr.GetIPv6(),
+		IPv4:        ipv4,
+		IPv6:        ipv6,
 	}
 
 	// call bgp sub-systems required to apply this policy's BGP topology.

--- a/pkg/bgpv1/agent/controller_test.go
+++ b/pkg/bgpv1/agent/controller_test.go
@@ -6,7 +6,7 @@ package agent_test
 import (
 	"context"
 	"errors"
-	"net"
+	"net/netip"
 	"testing"
 
 	"github.com/cilium/cilium/pkg/bgpv1/agent"
@@ -20,7 +20,7 @@ import (
 var (
 	// the standard node name we'll use throughout our tests.
 	nodeName = "node-under-test-01"
-	nodeIPv4 = net.ParseIP("192.168.0.1")
+	nodeIPv4 = netip.MustParseAddr("192.168.0.1")
 )
 
 // a mock agent.nodeSpecer implementation.
@@ -92,7 +92,7 @@ func TestControllerSanity(t *testing.T) {
 				if p != wantPolicy {
 					t.Fatalf("got: %+v, want: %+v", p, wantPolicy)
 				}
-				if !c.IPv4.Equal(nodeIPv4) {
+				if c.IPv4 != nodeIPv4 {
 					t.Fatalf("got: %v, want: %v", c.IPv4, nodeIPv4)
 				}
 				return nil
@@ -182,7 +182,7 @@ func TestControllerSanity(t *testing.T) {
 	}
 	for _, tt := range table {
 		t.Run(tt.name, func(t *testing.T) {
-			nodeaddr.SetIPv4(nodeIPv4)
+			nodeaddr.SetIPv4(nodeIPv4.AsSlice())
 			nodetypes.SetName(nodeName)
 			nodeSpecer := &fakeNodeSpecer{
 				Annotations_: tt.annotations,
@@ -202,7 +202,7 @@ func TestControllerSanity(t *testing.T) {
 			}
 			err := c.Reconcile(context.Background())
 			if (tt.err == nil) != (err == nil) {
-				t.Fatalf("wanted error: %v", tt.err == nil)
+				t.Fatalf("want: %v, got: %v", tt.err, err)
 			}
 		})
 	}

--- a/pkg/bgpv1/gobgp/server.go
+++ b/pkg/bgpv1/gobgp/server.go
@@ -6,7 +6,7 @@ package gobgp
 import (
 	"context"
 	"fmt"
-	"net"
+	"net/netip"
 
 	gobgp "github.com/osrg/gobgp/v3/api"
 	"github.com/osrg/gobgp/v3/pkg/server"
@@ -91,17 +91,16 @@ func NewGoBGPServerWithConfig(ctx context.Context, log *logrus.Entry, params typ
 // AddNeighbor will add the CiliumBGPNeighbor to the gobgp.BgpServer, creating
 // a BGP peering connection.
 func (g *GoBGPServer) AddNeighbor(ctx context.Context, n types.NeighborRequest) error {
-	// cilium neighbor uses CIDR string, gobgp neighbor uses IP string, convert.
-	var ip net.IP
-	var err error
-	if ip, _, err = net.ParseCIDR(n.Neighbor.PeerAddress); err != nil {
+	// cilium neighbor uses prefix string, gobgp neighbor uses IP string, convert.
+	prefix, err := netip.ParsePrefix(n.Neighbor.PeerAddress)
+	if err != nil {
 		// unlikely, we validate this on CR write to k8s api.
 		return fmt.Errorf("failed to parse PeerAddress: %w", err)
 	}
 	peerReq := &gobgp.AddPeerRequest{
 		Peer: &gobgp.Peer{
 			Conf: &gobgp.PeerConf{
-				NeighborAddress: ip.String(),
+				NeighborAddress: prefix.Addr().String(),
 				PeerAsn:         uint32(n.Neighbor.PeerASN),
 			},
 			// tells the peer we are capable of unicast IPv4 and IPv6
@@ -129,15 +128,14 @@ func (g *GoBGPServer) AddNeighbor(ctx context.Context, n types.NeighborRequest) 
 // RemoveNeighbor will remove the CiliumBGPNeighbor from the gobgp.BgpServer,
 // disconnecting the BGP peering connection.
 func (g *GoBGPServer) RemoveNeighbor(ctx context.Context, n types.NeighborRequest) error {
-	// cilium neighbor uses CIDR string, gobgp neighbor uses IP string, convert.
-	var ip net.IP
-	var err error
-	if ip, _, err = net.ParseCIDR(n.Neighbor.PeerAddress); err != nil {
+	// cilium neighbor uses prefix string, gobgp neighbor uses IP string, convert.
+	prefix, err := netip.ParsePrefix(n.Neighbor.PeerAddress)
+	if err != nil {
 		// unlikely, we validate this on CR write to k8s api.
 		return fmt.Errorf("failed to parse PeerAddress: %w", err)
 	}
 	peerReq := &gobgp.DeletePeerRequest{
-		Address: ip.String(),
+		Address: prefix.Addr().String(),
 	}
 	if err := g.server.DeletePeer(ctx, peerReq); err != nil {
 		return fmt.Errorf("failed while reconciling neighbor %v %v: %w", n.Neighbor.PeerAddress, n.Neighbor.PeerASN, err)
@@ -163,17 +161,16 @@ func (g *GoBGPServer) AdvertisePath(ctx context.Context, p types.PathRequest) (t
 	var err error
 	var path *gobgp.Path
 	var resp *gobgp.AddPathResponse
-	ip := p.Advert.Net
+	prefix := p.Advert.Prefix
 
 	origin, _ := apb.New(&gobgp.OriginAttribute{
 		Origin: 0,
 	})
 	switch {
-	case ip.IP.To4() != nil:
-		prefixLen, _ := ip.Mask.Size()
+	case prefix.Addr().Is4():
 		nlri, _ := apb.New(&gobgp.IPAddressPrefix{
-			PrefixLen: uint32(prefixLen),
-			Prefix:    ip.IP.String(),
+			PrefixLen: uint32(prefix.Bits()),
+			Prefix:    prefix.Addr().String(),
 		})
 		// Currently, we only support advertising locally originated paths (the paths generated in Cilium
 		// node itself, not the paths received from another BGP Peer or redistributed from another routing
@@ -201,11 +198,10 @@ func (g *GoBGPServer) AdvertisePath(ctx context.Context, p types.PathRequest) (t
 		resp, err = g.server.AddPath(ctx, &gobgp.AddPathRequest{
 			Path: path,
 		})
-	case ip.IP.To16() != nil:
-		prefixLen, _ := ip.Mask.Size()
+	case prefix.Addr().Is6():
 		nlri, _ := apb.New(&gobgp.IPAddressPrefix{
-			PrefixLen: uint32(prefixLen),
-			Prefix:    ip.IP.String(),
+			PrefixLen: uint32(prefix.Bits()),
+			Prefix:    prefix.Addr().String(),
 		})
 		nlriAttrs, _ := apb.New(&gobgp.MpReachNLRIAttribute{ // MP BGP NLRI
 			Family: GoBGPIPv6Family,
@@ -222,14 +218,14 @@ func (g *GoBGPServer) AdvertisePath(ctx context.Context, p types.PathRequest) (t
 			Path: path,
 		})
 	default:
-		return types.PathResponse{}, fmt.Errorf("provided IP returned nil for both IPv4 and IPv6 lengths: %v", len(ip.IP))
+		return types.PathResponse{}, fmt.Errorf("unknown address family for prefix %s", prefix.String())
 	}
 	if err != nil {
 		return types.PathResponse{}, err
 	}
 	return types.PathResponse{
 		Advert: types.Advertisement{
-			Net:           ip,
+			Prefix:        prefix,
 			GoBGPPathUUID: resp.Uuid,
 		},
 	}, err

--- a/pkg/bgpv1/manager/advertisements_reconciler.go
+++ b/pkg/bgpv1/manager/advertisements_reconciler.go
@@ -6,7 +6,7 @@ package manager
 import (
 	"context"
 	"fmt"
-	"net"
+	"net/netip"
 
 	"github.com/sirupsen/logrus"
 
@@ -24,12 +24,12 @@ type advertisementsReconcilerParams struct {
 	newc *v2alpha1api.CiliumBGPVirtualRouter
 
 	currentAdvertisements []types.Advertisement
-	toAdvertise           []*net.IPNet
+	toAdvertise           []netip.Prefix
 }
 
 // exportAdvertisementsReconciler reconciles the state of the BGP advertisements
-// with the provided toAdvertise list of IPNets and returns a list of the
-// advertisements currently being announced.
+// with the provided toAdvertise list and returns a list of the advertisements
+// currently being announced.
 func exportAdvertisementsReconciler(params *advertisementsReconcilerParams) ([]types.Advertisement, error) {
 	var (
 		l = log.WithFields(
@@ -55,7 +55,7 @@ func exportAdvertisementsReconciler(params *advertisementsReconcilerParams) ([]t
 		l.Debugf("%s advertisements disabled for virtual router with local ASN %v", params.name, params.newc.LocalASN)
 
 		for _, advrt := range params.currentAdvertisements {
-			l.Debugf("Withdrawing %s advertisement %v for local ASN %v", params.name, advrt.Net.String(), params.newc.LocalASN)
+			l.Debugf("Withdrawing %s advertisement %v for local ASN %v", params.name, advrt.Prefix.String(), params.newc.LocalASN)
 			if err := params.sc.Server.WithdrawPath(params.ctx, types.PathRequest{Advert: advrt}); err != nil {
 				return nil, err
 			}
@@ -74,18 +74,18 @@ func exportAdvertisementsReconciler(params *advertisementsReconcilerParams) ([]t
 	aset := map[string]*member{}
 
 	// populate the advrts that must be present, universe a
-	for _, ipNet := range params.toAdvertise {
+	for _, prefix := range params.toAdvertise {
 		var (
 			m  *member
 			ok bool
 		)
 
-		key := ipNet.String()
+		key := prefix.String()
 		if m, ok = aset[key]; !ok {
 			aset[key] = &member{
 				a: true,
 				advrt: &types.Advertisement{
-					Net: ipNet,
+					Prefix: prefix,
 				},
 			}
 			continue
@@ -99,12 +99,12 @@ func exportAdvertisementsReconciler(params *advertisementsReconcilerParams) ([]t
 			m  *member
 			ok bool
 		)
-		key := advrt.Net.String()
+		key := advrt.Prefix.String()
 		if m, ok = aset[key]; !ok {
 			aset[key] = &member{
 				b: true,
 				advrt: &types.Advertisement{
-					Net:           advrt.Net,
+					Prefix:        advrt.Prefix,
 					GoBGPPathUUID: advrt.GoBGPPathUUID,
 				},
 			}
@@ -138,17 +138,17 @@ func exportAdvertisementsReconciler(params *advertisementsReconcilerParams) ([]t
 
 	// create new adverts
 	for _, advrt := range toAdvertise {
-		l.Debugf("Advertising %s %v for policy with local ASN: %v", params.name, advrt.Net.String(), params.newc.LocalASN)
+		l.Debugf("Advertising %s %v for policy with local ASN: %v", params.name, advrt.Prefix.String(), params.newc.LocalASN)
 		advrtResp, err := params.sc.Server.AdvertisePath(params.ctx, types.PathRequest{Advert: advrt})
 		if err != nil {
-			return nil, fmt.Errorf("failed to advertise %s prefix %v: %w", params.name, advrt.Net, err)
+			return nil, fmt.Errorf("failed to advertise %s prefix %v: %w", params.name, advrt.Prefix, err)
 		}
 		newAdverts = append(newAdverts, advrtResp.Advert)
 	}
 
 	// withdraw uneeded adverts
 	for _, advrt := range toWithdraw {
-		l.Debugf("Withdrawing %s %v for policy with local ASN: %v", params.name, advrt.Net, params.newc.LocalASN)
+		l.Debugf("Withdrawing %s %v for policy with local ASN: %v", params.name, advrt.Prefix, params.newc.LocalASN)
 		if err := params.sc.Server.WithdrawPath(params.ctx, types.PathRequest{Advert: advrt}); err != nil {
 			return nil, err
 		}

--- a/pkg/bgpv1/manager/manager.go
+++ b/pkg/bgpv1/manager/manager.go
@@ -6,7 +6,6 @@ package manager
 import (
 	"context"
 	"fmt"
-	"net"
 	"sort"
 
 	"github.com/cilium/cilium/api/v1/models"
@@ -211,22 +210,9 @@ func (m *BGPRouterManager) registerBGPServer(ctx context.Context, c *v2alpha1api
 		}
 	}
 
-	// resolve router ID, if we have an annotation and it can be parsed into
-	// a valid ipv4 address use this,
-	//
-	// if not determine if Cilium is configured with an IPv4 address, if so use
-	// this.
-	//
-	// if neither, return an error, we cannot assign an router ID.
-	var routerID string
-	_, ok := cstate.Annotations[c.LocalASN]
-	switch {
-	case ok && !net.ParseIP(cstate.Annotations[c.LocalASN].RouterID).IsUnspecified():
-		routerID = cstate.Annotations[c.LocalASN].RouterID
-	case !cstate.IPv4.IsUnspecified():
-		routerID = cstate.IPv4.String()
-	default:
-		return fmt.Errorf("router id not specified by annotation and no IPv4 address assigned by cilium, cannot resolve router id for virtual router with local ASN %v", c.LocalASN)
+	routerID, err := cstate.ResolveRouterID(c.LocalASN)
+	if err != nil {
+		return err
 	}
 
 	globalConfig := types.ServerParameters{

--- a/pkg/bgpv1/manager/reconcile.go
+++ b/pkg/bgpv1/manager/reconcile.go
@@ -6,7 +6,7 @@ package manager
 import (
 	"context"
 	"fmt"
-	"net"
+	"net/netip"
 
 	"github.com/cilium/cilium/pkg/bgpv1/agent"
 	"github.com/cilium/cilium/pkg/bgpv1/types"
@@ -118,22 +118,9 @@ func preflightReconciler(
 		}
 	}
 
-	// resolve router ID, if we have an annotation and it can be parsed into
-	// a valid ipv4 address use this,
-	//
-	// if not determine if Cilium is configured with an IPv4 address, if so use
-	// this.
-	//
-	// if neither, return an error, we cannot assign an router ID.
-	var routerID string
-	_, ok := cstate.Annotations[newc.LocalASN]
-	switch {
-	case ok && !net.ParseIP(cstate.Annotations[newc.LocalASN].RouterID).IsUnspecified():
-		routerID = cstate.Annotations[newc.LocalASN].RouterID
-	case !cstate.IPv4.IsUnspecified():
-		routerID = cstate.IPv4.String()
-	default:
-		return fmt.Errorf("router id not specified by annotation and no IPv4 address assigned by cilium, cannot resolve router id for virtual router with local ASN %v", newc.LocalASN)
+	routerID, err := cstate.ResolveRouterID(newc.LocalASN)
+	if err != nil {
+		return err
 	}
 
 	var shouldRecreate bool
@@ -360,13 +347,13 @@ func exportPodCIDRReconciler(
 		return fmt.Errorf("attempted pod CIDR advertisements reconciliation with nil ControlPlaneState")
 	}
 
-	toAdvertise := []*net.IPNet{}
+	toAdvertise := []netip.Prefix{}
 	for _, cidr := range cstate.PodCIDRs {
-		_, ipNet, err := net.ParseCIDR(cidr)
+		prefix, err := netip.ParsePrefix(cidr)
 		if err != nil {
-			return fmt.Errorf("failed to parse cidr %s: %w", cidr, err)
+			return fmt.Errorf("failed to parse prefix %s: %w", cidr, err)
 		}
-		toAdvertise = append(toAdvertise, ipNet)
+		toAdvertise = append(toAdvertise, prefix)
 	}
 
 	advertisements, err := exportAdvertisementsReconciler(&advertisementsReconcilerParams{
@@ -518,7 +505,7 @@ func (r *LBServiceReconciler) svcDiffReconciliation(ctx context.Context, sc *Ser
 
 // svcDesiredRoutes determines which, if any routes should be announced for the given service. This determines the
 // desired state.
-func (r *LBServiceReconciler) svcDesiredRoutes(newc *v2alpha1api.CiliumBGPVirtualRouter, svc *slim_corev1.Service) ([]*net.IPNet, error) {
+func (r *LBServiceReconciler) svcDesiredRoutes(newc *v2alpha1api.CiliumBGPVirtualRouter, svc *slim_corev1.Service) ([]netip.Prefix, error) {
 	if newc.ServiceSelector == nil {
 		// If the vRouter has no service selector, there are no desired routes
 		return nil, nil
@@ -539,22 +526,18 @@ func (r *LBServiceReconciler) svcDesiredRoutes(newc *v2alpha1api.CiliumBGPVirtua
 		return nil, nil
 	}
 
-	var desiredRoutes []*net.IPNet
+	var desiredRoutes []netip.Prefix
 	for _, ingress := range svc.Status.LoadBalancer.Ingress {
 		if ingress.IP == "" {
 			continue
 		}
 
-		cidr := &net.IPNet{
-			IP: net.ParseIP(ingress.IP),
-		}
-		if cidr.IP.To4() == nil {
-			cidr.Mask = net.CIDRMask(128, 128)
-		} else {
-			cidr.Mask = net.CIDRMask(32, 32)
+		addr, err := netip.ParseAddr(ingress.IP)
+		if err != nil {
+			continue
 		}
 
-		desiredRoutes = append(desiredRoutes, cidr)
+		desiredRoutes = append(desiredRoutes, netip.PrefixFrom(addr, addr.BitLen()))
 	}
 
 	return desiredRoutes, err
@@ -573,7 +556,7 @@ func (r *LBServiceReconciler) reconcileService(ctx context.Context, sc *ServerWi
 	for _, desiredCidr := range desiredCidrs {
 		// If this route has already been announced, don't add it again
 		if slices.IndexFunc(sc.ServiceAnnouncements[svcKey], func(existing types.Advertisement) bool {
-			return cidrEqual(desiredCidr, existing.Net)
+			return desiredCidr == existing.Prefix
 		}) != -1 {
 			continue
 		}
@@ -581,7 +564,7 @@ func (r *LBServiceReconciler) reconcileService(ctx context.Context, sc *ServerWi
 		// Advertise the new cidr
 		advertPathResp, err := sc.Server.AdvertisePath(ctx, types.PathRequest{
 			Advert: types.Advertisement{
-				Net: desiredCidr,
+				Prefix: desiredCidr,
 			},
 		})
 		if err != nil {
@@ -594,8 +577,8 @@ func (r *LBServiceReconciler) reconcileService(ctx context.Context, sc *ServerWi
 	for i := len(sc.ServiceAnnouncements[svcKey]) - 1; i >= 0; i-- {
 		announcement := sc.ServiceAnnouncements[svcKey][i]
 		// If the announcement is within the list of desired routes, don't remove it
-		if slices.IndexFunc(desiredCidrs, func(existing *net.IPNet) bool {
-			return cidrEqual(existing, announcement.Net)
+		if slices.IndexFunc(desiredCidrs, func(existing netip.Prefix) bool {
+			return existing == announcement.Prefix
 		}) != -1 {
 			continue
 		}
@@ -620,7 +603,7 @@ func (r *LBServiceReconciler) withdrawService(ctx context.Context, sc *ServerWit
 		if err := sc.Server.WithdrawPath(ctx, types.PathRequest{Advert: advertisement}); err != nil {
 			// Persist remaining advertisements
 			sc.ServiceAnnouncements[key] = advertisements
-			return fmt.Errorf("failed to withdraw deleted service route: %v: %w", advertisement.Net, err)
+			return fmt.Errorf("failed to withdraw deleted service route: %v: %w", advertisement.Prefix, err)
 		}
 
 		// Delete the advertisement after each withdraw in case we error half way through
@@ -631,12 +614,6 @@ func (r *LBServiceReconciler) withdrawService(ctx context.Context, sc *ServerWit
 	delete(sc.ServiceAnnouncements, key)
 
 	return nil
-}
-
-func cidrEqual(a, b *net.IPNet) bool {
-	aOnes, aSize := a.Mask.Size()
-	bOnes, bSize := b.Mask.Size()
-	return a.IP.Equal(b.IP) && aOnes == bOnes && aSize == bSize
 }
 
 func serviceLabelSet(svc *slim_corev1.Service) labels.Labels {

--- a/pkg/bgpv1/manager/reconcile_test.go
+++ b/pkg/bgpv1/manager/reconcile_test.go
@@ -5,7 +5,7 @@ package manager
 
 import (
 	"context"
-	"net"
+	"net/netip"
 	"testing"
 
 	"github.com/stretchr/testify/require"
@@ -270,14 +270,11 @@ func TestNeighborReconciler(t *testing.T) {
 			}
 
 			for _, n := range tt.newNeighbors {
-				ip, _, err := net.ParseCIDR(n)
-				if err != nil {
-					t.Fatalf("failed to parse neighbor ip: %v", err)
-				}
+				prefix := netip.MustParsePrefix(n)
 				var seen bool
 				for _, p := range peers {
-					ipp := net.ParseIP(p.PeerAddress)
-					if ip.Equal(ipp) {
+					addr := netip.MustParseAddr(p.PeerAddress)
+					if prefix.Addr() == addr {
 						seen = true
 					}
 				}
@@ -287,14 +284,11 @@ func TestNeighborReconciler(t *testing.T) {
 			}
 
 			for _, p := range peers {
-				ip := net.ParseIP(p.PeerAddress)
+				paddr := netip.MustParseAddr(p.PeerAddress)
 				var seen bool
 				for _, n := range tt.newNeighbors {
-					ipp, _, err := net.ParseCIDR(n)
-					if err != nil {
-						t.Fatalf("failed to parse peer ip: %v", err)
-					}
-					if ip.Equal(ipp) {
+					addr := netip.MustParsePrefix(n)
+					if paddr == addr.Addr() {
 						seen = true
 					}
 				}
@@ -317,7 +311,7 @@ func TestExportPodCIDRReconciler(t *testing.T) {
 		// the advertised PodCIDR blocks the test begins with, these are encoded
 		// into Golang structs for the convenience of passing directly to the
 		// ServerWithConfig.AdvertisePath() method.
-		advertised []*net.IPNet
+		advertised []netip.Prefix
 		// the updated PodCIDR blocks to reconcile, these are string encoded
 		// for the convenience of attaching directly to the NodeSpec.PodCIDRs
 		// field.
@@ -329,11 +323,8 @@ func TestExportPodCIDRReconciler(t *testing.T) {
 			name:         "disable",
 			enabled:      true,
 			shouldEnable: false,
-			advertised: []*net.IPNet{
-				{
-					IP:   net.ParseIP("192.168.0.0"),
-					Mask: net.IPv4Mask(255, 255, 255, 0),
-				},
+			advertised: []netip.Prefix{
+				netip.MustParsePrefix("192.168.0.0/24"),
 			},
 		},
 		{
@@ -346,11 +337,8 @@ func TestExportPodCIDRReconciler(t *testing.T) {
 			name:         "no change",
 			enabled:      true,
 			shouldEnable: true,
-			advertised: []*net.IPNet{
-				{
-					IP:   net.ParseIP("192.168.0.0"),
-					Mask: net.IPv4Mask(255, 255, 255, 0),
-				},
+			advertised: []netip.Prefix{
+				netip.MustParsePrefix("192.168.0.0/24"),
 			},
 			updated: []string{"192.168.0.0/24"},
 		},
@@ -358,11 +346,8 @@ func TestExportPodCIDRReconciler(t *testing.T) {
 			name:         "additional network",
 			enabled:      true,
 			shouldEnable: true,
-			advertised: []*net.IPNet{
-				{
-					IP:   net.ParseIP("192.168.0.0"),
-					Mask: net.IPv4Mask(255, 255, 255, 0),
-				},
+			advertised: []netip.Prefix{
+				netip.MustParsePrefix("192.168.0.0/24"),
 			},
 			updated: []string{"192.168.0.0/24", "192.168.1.0/24"},
 		},
@@ -370,15 +355,9 @@ func TestExportPodCIDRReconciler(t *testing.T) {
 			name:         "removal of both networks",
 			enabled:      true,
 			shouldEnable: true,
-			advertised: []*net.IPNet{
-				{
-					IP:   net.ParseIP("192.168.0.0"),
-					Mask: net.IPv4Mask(255, 255, 255, 0),
-				},
-				{
-					IP:   net.ParseIP("192.168.1.0"),
-					Mask: net.IPv4Mask(255, 255, 255, 0),
-				},
+			advertised: []netip.Prefix{
+				netip.MustParsePrefix("192.168.0.0/24"),
+				netip.MustParsePrefix("192.168.1.0/24"),
 			},
 			updated: []string{},
 		},
@@ -408,7 +387,7 @@ func TestExportPodCIDRReconciler(t *testing.T) {
 			for _, cidr := range tt.advertised {
 				advrtResp, err := testSC.Server.AdvertisePath(context.Background(), types.PathRequest{
 					Advert: types.Advertisement{
-						Net: cidr,
+						Prefix: cidr,
 					},
 				})
 				if err != nil {
@@ -424,7 +403,7 @@ func TestExportPodCIDRReconciler(t *testing.T) {
 			}
 			newcstate := agent.ControlPlaneState{
 				PodCIDRs: tt.updated,
-				IPv4:     net.ParseIP("127.0.0.1"),
+				IPv4:     netip.MustParseAddr("127.0.0.1"),
 			}
 
 			err = exportPodCIDRReconciler(context.Background(), testSC, newc, &newcstate)
@@ -444,13 +423,10 @@ func TestExportPodCIDRReconciler(t *testing.T) {
 
 			// ensure we see tt.updated in testSC.PodCIDRAnnoucements
 			for _, cidr := range tt.updated {
-				_, parsed, err := net.ParseCIDR(cidr)
-				if err != nil {
-					t.Fatalf("failed to parse updated cidr: %v", err)
-				}
+				prefix := netip.MustParsePrefix(cidr)
 				var seen bool
 				for _, advrt := range testSC.PodCIDRAnnouncements {
-					if advrt.Net.String() == parsed.String() {
+					if advrt.Prefix == prefix {
 						seen = true
 					}
 				}
@@ -464,11 +440,7 @@ func TestExportPodCIDRReconciler(t *testing.T) {
 			for _, advrt := range testSC.PodCIDRAnnouncements {
 				var seen bool
 				for _, cidr := range tt.updated {
-					_, parsed, err := net.ParseCIDR(cidr)
-					if err != nil {
-						t.Fatalf("failed to parse updated cidr: %v", err)
-					}
-					if advrt.Net.String() == parsed.String() {
+					if advrt.Prefix == netip.MustParsePrefix(cidr) {
 						seen = true
 					}
 				}
@@ -905,13 +877,10 @@ func TestLBServiceReconciler(t *testing.T) {
 			testSC.Config = oldc
 			for svcKey, cidrs := range tt.advertised {
 				for _, cidr := range cidrs {
-					_, net, err := net.ParseCIDR(cidr)
-					if err != nil {
-						t.Fatalf("bad cidr '%s': %s", cidr, err)
-					}
+					prefix := netip.MustParsePrefix(cidr)
 					advrtResp, err := testSC.Server.AdvertisePath(context.Background(), types.PathRequest{
 						Advert: types.Advertisement{
-							Net: net,
+							Prefix: prefix,
 						},
 					})
 					if err != nil {
@@ -928,7 +897,7 @@ func TestLBServiceReconciler(t *testing.T) {
 				ServiceSelector: tt.newServiceSelector,
 			}
 			newcstate := agent.ControlPlaneState{
-				IPv4: net.ParseIP("127.0.0.1"),
+				IPv4: netip.MustParseAddr("127.0.0.1"),
 			}
 
 			diffstore := newFakeDiffStore[*slim_corev1.Service]()
@@ -962,13 +931,10 @@ func TestLBServiceReconciler(t *testing.T) {
 			// ensure we see tt.updated in testSC.ServiceAnnouncements
 			for svcKey, cidrs := range tt.updated {
 				for _, cidr := range cidrs {
-					_, parsed, err := net.ParseCIDR(cidr)
-					if err != nil {
-						t.Fatalf("failed to parse updated cidr: %v", err)
-					}
+					prefix := netip.MustParsePrefix(cidr)
 					var seen bool
 					for _, advrt := range testSC.ServiceAnnouncements[svcKey] {
-						if advrt.Net.String() == parsed.String() {
+						if advrt.Prefix == prefix {
 							seen = true
 						}
 					}
@@ -984,11 +950,7 @@ func TestLBServiceReconciler(t *testing.T) {
 				for _, advrt := range advrts {
 					var seen bool
 					for _, cidr := range tt.updated[svcKey] {
-						_, parsed, err := net.ParseCIDR(cidr)
-						if err != nil {
-							t.Fatalf("failed to parse updated cidr: %v", err)
-						}
-						if advrt.Net.String() == parsed.String() {
+						if advrt.Prefix == netip.MustParsePrefix(cidr) {
 							seen = true
 						}
 					}

--- a/pkg/bgpv1/types/bgp.go
+++ b/pkg/bgpv1/types/bgp.go
@@ -5,7 +5,7 @@ package types
 
 import (
 	"context"
-	"net"
+	"net/netip"
 
 	"github.com/cilium/cilium/api/v1/models"
 	v2alpha1api "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
@@ -25,15 +25,15 @@ type RouteSelectionOptions struct {
 	AdvertiseInactiveRoutes bool
 }
 
-// Advertisement is a container object which associates a net.IPNet
+// Advertisement is a container object which associates a netip.Prefix
 //
-// The `Net` field makes comparing this Advertisement with another IPNet encoded
+// The `Prefix` field makes comparing this Advertisement with another Prefix encoded
 // prefixes simple.
 //
 // The `GoBGPPathUUID` field is a gobgp.AddPathResponse.Uuid object which can be forwarded to gobgp's
 // WithdrawPath method, making withdrawing an advertised route simple.
 type Advertisement struct {
-	Net           *net.IPNet
+	Prefix        netip.Prefix
 	GoBGPPathUUID []byte // path identifier in underlying implementation
 }
 

